File: queue/arrayqueue/python/test_queue.py
#!/usr/bin/env python3

from arrayqueue import ArrayQueue as Queue

def test_queue1():
    myqueue = Queue()
    sum = 0
    for i in range(1, 11):
        myqueue.add(i)

    while not myqueue.isEmpty():
        sum += myqueue.pop()

    print("total:", sum)

def test_queue2():
    myints = Queue()
    print("0. size:", len(myints))

    for i in range(5):
        myints.add(i)
    print("1. size:", len(myints))

    myints.pop()
    print("2. size:", len(myints))

if __name__ == "__main__":
    test_queue1()
    test_queue2()

File: queue/arrayqueue/benchmark-cxx/test_queue.cpp
#include <queue>
#include <iostream>

void test_queue1() {
    std::queue<int> myqueue;
    int sum (0);

    for (int i=1;i<=10;i++) myqueue.push(i);

    while (!myqueue.empty())
    {
        sum += myqueue.front();
        myqueue.pop();
    }

    std::cout << "total: " << sum << '\n';
}

void test_queue2() {
    std::queue<int> myints;
    std::cout << "0. size: " << myints.size() << '\n';

    for (int i=0; i<5; i++) myints.push(i);
    std::cout << "1. size: " << myints.size() << '\n';

    myints.pop();
    std::cout << "2. size: " << myints.size() << '\n';
}

int main() {
    test_queue1();
}
